### PR TITLE
11.0 mig partner contact birthdate

### PR DESCRIPTION
**Pull request by jprobineau**

### CODE REVIEW

**Comment by yajo** [comment]
Setting as WIP until partner_contact_personal_information_page is merged

**Comment by jprobineau** [comment]
Module partner_contact_personal_information_page is in issue #511

**Comment by simoneversienti** [comment]
Ok :+1:

File: partner_contact_birthdate/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    # New birthdate field in date format
+    birthdate_date = fields.Date("Birthdate")
+
+    # Make the old Char field to reflect the new Date field
+    birthdate = fields.Char(
+        compute="_birthdate_compute",
+        inverse="_birthdate_inverse",
+        store=True)
+
+    @api.one
```
Comment (OpenCode) [inline partner_contact_birthdate/models/res_partner.py:24]: Why are you using api.one and not api.multi?

File: partner_contact_birthdate/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @api.depends("birthdate_date")
+    def _birthdate_compute(self):
+        """Store a string of the new date in the old field."""
+        self.birthdate = self.birthdate_date
+
+    @api.one
```
Comment (OpenCode) [inline partner_contact_birthdate/models/res_partner.py:30]: Why are you using api.one and not api.multi?

**Review by OpenCode — CHANGES_REQUESTED** [review]

File: partner_contact_birthdate/__init__.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
-# -*- coding: utf-8 -*-
 # Copyright (C) 2014-2015  Grupo ESOC <www.grupoesoc.es>
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# © 2017-Apertoso N.V. (<http://www.apertoso.be>)
```
Comment (lasley) [inline partner_contact_birthdate/__init__.py]: This change does not warrant the addition of your copyright IMO. Same goes for all the files.

File: partner_contact_birthdate/models/res_partner.py (hunk 1 above)
Comment (lasley) [reply partner_contact_birthdate/models/res_partner.py:24]: We could add `oca-decorators` to the python dependencies, then switch to `oca.decorators.foreach`

**Comment by pedrobaeza** [comment]
Are you going to work on this PR anymore? We are in the code sprint and we need it as dependency

**Comment by ghost** [comment]
error tells partner_contact_personal_information_page isnt available, but it is in the repo?

File: partner_contact_birthdate/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+class ResPartner(models.Model):
+    """Partner with birth date in date format."""
+    _inherit = "res.partner"
+
+    # New birthdate field in date format
+    birthdate_date = fields.Date("Birthdate")
+
+    # Make the old Char field to reflect the new Date field
+    birthdate = fields.Char(
```
Comment (Olivier-LAURENT) [inline partner_contact_birthdate/models/res_partner.py:19]: Hi @jprobineau , In V11 Odoo has removed this field. Is there any interest to reintroduce it here?

**Comment by pedrobaeza** [comment]
Superseeded by #617

**Pull request closed without merging**